src/inmanta/db/versions/v202503030.py

```diff
@@ -0,0 +1,238 @@
+        resource_type_to_module: dict[int, dict[str, set[str]]],
+    ) -> None:
+        fetch_source_refs_query = """
+    SELECT DISTINCT environment, version, source_refs, resource from public.code
+        """
+        result = await connection.fetch(fetch_source_refs_query)
+        for res in result:
+            env: str = str(res["environment"])
+            model_version = res["version"]
+            source_refs = res["source_refs"]
+            resource_type: str = str(res["resource"])
+
+            assert isinstance(model_version, int)
+
+            for file_hash, file_data in source_refs.items():  # type: ignore
+                file_path, python_module_name, requirements = file_data
+                inmanta_module_name = python_module_name.split(".")[1]
+                assert isinstance(inmanta_module_name, str)
+                code_data[env][model_version][inmanta_module_name].add(
+                    tuple((file_hash, file_path, python_module_name, frozenset(requirements)))
+                )
+
+                resource_type_to_module[model_version][resource_type].add(inmanta_module_name)
+
+    def build_module_data(
+        code_data: dict[str, dict[int, dict[str, set[tuple[str, str, str, set[str]]]]]],
+        module_data: list[tuple[str, str, str, list[str]]],
+        files_in_module_data: list[tuple[str, str, str, str, str]],
+        model_to_module_version_map: dict[int, dict[str, str]],
+    ) -> None:
+
+        def compute_version_requirements(
+            source_info: set[tuple[str, str, str, set[str]]],
+        ) -> tuple[str, list[str]]:
+            """
+            Helper method
+            """
+            reqs = set()
+            module_version_hash = hashlib.new("sha1")
+
+            for file_hash, _, _, requirements in sorted(source_info, key=lambda x: x[0]):
+                # sort by hash to compute inmanta module version
+                module_version_hash.update(file_hash.encode())
+                reqs.update(requirements)
+
+            module_version = module_version_hash.hexdigest()
+            return module_version, list(reqs)
+
+        def compute_files_in_module(
+            source_info: set[tuple[str, str, str, set[str]]],
+            module_name: str,
+            environment: str,
+            module_version: str,
+            files_in_module_data: list[tuple[str, str, str, str, str]],
+        ) -> None:
+            """
+            Helper method
+            """
+            for file_hash, file_path, _, _ in source_info:
+                parts = Path(file_path).parts
+                if f"{module_name}/plugins" in file_path:
+                    # V1 module
+                    rel_py_file = os.path.relpath(file_path, start="plugins")
+                    relative_path = os.path.join(module_name, loader.PLUGIN_DIR, rel_py_file)
+                else:
+                    relative_path = str(Path(*parts[parts.index(const.PLUGINS_PACKAGE) :]))
```

Comment:
I wonder if we need this and if the correct, normalized form is already available in one of the other arguments

Reply:
Indeed, updated this to use the `python_module_name` argument. I had some misconceptions about how it behaved for v1 vs v2 modules. It will actually always be of the form `inmanta_plugins.<inmanta_module_name>.<python_module_name>` regardless of the module's version.